apps/confessions-server/src/routing/routes/confessionRoutes.ts

```diff
@@ -0,0 +1,117 @@
+import express, { Request, Response, NextFunction } from "express";
+import {
+  SemaphoreGroupPCDPackage,
+  SerializedSemaphoreGroup
+} from "@pcd/semaphore-group-pcd";
+import { Group } from "@semaphore-protocol/group";
+import { sha256 } from "js-sha256";
+import { ApplicationContext } from "../../types";
+import { prisma } from "../../util/prisma";
+import { IS_PROD } from "../../util/isProd";
+
+const SEMAPHORE_GROUP_URL = IS_PROD
+  ? "https://api.pcd-passport.com/semaphore/1"
+  : "http://localhost:3002/semaphore/1";
+
+export function initConfessionRoutes(
+  app: express.Application,
+  _context: ApplicationContext
+): void {
+  app.post("/confessions", async (req: Request, res: Response, next: NextFunction) => {
+    const request = req.body as PostConfessionRequest;
+
+    try {
+      // only allow Zuzalu group for now
+      if (request.semaphoreGroupUrl !== SEMAPHORE_GROUP_URL) {
+        throw new Error("only Zuzalu group is allowed");
+      }
+
+      // verify confession proof
+      const pcd = await SemaphoreGroupPCDPackage.deserialize(
+        request.proof
+      );
+
+      const verified = await SemaphoreGroupPCDPackage.verify(pcd);
+      if (!verified) {
+        throw new Error("invalid proof");
+      }
+
+      // check semaphoreGoupUrl and confession in the request matches the proof
+      const response = await fetch(request.semaphoreGroupUrl);
+      const json = await response.text();
+      const serializedGroup = JSON.parse(json) as SerializedSemaphoreGroup;
+      const group = new Group(1, 16);
+      group.addMembers(serializedGroup.members);
+      if (pcd.proof.proof.merkleTreeRoot !== group.root.toString()) {
```

Comment:
@ichub I realized that ideally I should be using `pcd.claim` here rather than `proof` (as how I check the signal below). Ideally, third party apps should not be using what's inside proof directly (even inside the integration).
However in this case, it is easier to use the `merrkleTreeRoot` in the proof. And I copied this code from here: https://github.com/proofcarryingdata/zupass/blob/main/packages/passport-interface/src/SemaphoreGroupIntegration.ts#L92-L105.
Wondering how you think about this.

Reply:
cool, will use it once your PR gets merged

Reply:
changed to use `pcd.claim`